sdk/identity/identity/test/internal/node/managedIdentityCredential/imdsRetryPolicy.spec.ts

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { imdsRetryPolicy } from "../../../../src/credentials/managedIdentityCredential/imdsRetryPolicy";
+import { PipelineRequest, SendRequest, createHttpHeaders } from "@azure/core-rest-pipeline";
+import { MSIConfiguration } from "../../../../src/credentials/managedIdentityCredential/models";
+import { assert } from "@azure-tools/test-utils";
+import { IdentityClient } from "../../../../src/client/identityClient";
+
+describe("imdsRetryPolicy", () => {
+  const mockRetryConfig: MSIConfiguration["retryConfig"] = {
+    maxRetries: 3,
+    startDelayInMs: 100,
+    intervalIncrement: 1000,
```

Comment:
are we actually waiting in this test? Should we mock the timer?

Reply:
We are waiting in this test, but rather than mocking the timers I went with a super low timeout - basically low enough that I don't get `yellow` or `red` timer output in the tests 😄 https://github.com/Azure/azure-sdk-for-js/pull/30172/files#diff-effa0ad08c76ebd01e72e27643ff1844fccc920e32f58fd96bc1b6da86b63c5bR16-R20

I can mock timers but I think this is ok now as the test runs fast but if you feel strongly then I can definitely look at alternatives